### PR TITLE
Authenticate operations

### PR DESCRIPTION
**Pull request by pilhuhn**
With Hawkular-services, access to operations ('command-gw') needs to be authenticated, otherwise they will just fail.

### CODE REVIEW

File: lib/hawkular/operations/operations_api.rb [hunk 1]
```diff
@@ -53,14 +53,29 @@ def to_msg_hash
     # @example
     #   Hawkular::Operations::OperationsClient.new(credentials: {username: 'jdoe', password: 'password'})
     def initialize(args)
-      args[:host] ||= 'localhost:8080'
+      ep = args[:entrypoint]
```
Comment (abonas) [inline lib/hawkular/operations/operations_api.rb:59]: btw I'm confused where is entrypoint coming from? I read lines 44-51 and don't see it there

Comment (pilhuhn) [inline lib/hawkular/operations/operations_api.rb:59]: On 27 Jun 2016, at 12:08, abonas wrote:

> > @@ -53,14 +53,29 @@ def to_msg_hash
> >      # @example
> >      #   Hawkular::Operations::OperationsClient.new(credentials: 
> > {username: 'jdoe', password: 'password'})
> >      def initialize(args)
> > -      args[:host] ||= 'localhost:8080'
> > -      ep = args[:entrypoint]
> 
> btw I'm confused where is entrypoint coming from? I read lines 44-51 
> and don't see it there

I'll add it to the javadoc.

It is passed from here 
https://github.com/pilhuhn/hawkular-client-ruby/blob/f27efffd42de858a9fc1238dcc693a62abd2b7e2/lib/hawkular/hawkular_client.rb#L61
when one does

```
client = Hawkular::Client.new()
ops_client = client.operations()
```

As this `init_operations_client()` never passed `args[:host]` the 
default of `localhost:8080` kicked in :-(

Comment (abonas) [inline lib/hawkular/operations/operations_api.rb:59]: ok, thanks
there's no javadoc in a ruby client :wink: 
the lack of proper client documentation is really an obstacle...

File: spec/integration/operations_spec.rb [hunk 2]
```diff
@@ -20,16 +20,96 @@ module Hawkular::Operations::RSpec
...
+      end
+    end
+
+    it 'should bail with no host' do
+      WebSocketVCR.configure do |c|
+        c.hook_uris = [HOST]
+      end
+
+      WebSocketVCR.record(example, self) do
+        # rubocop:disable Lint/HandleExceptions
```
Comment (abonas) [inline spec/integration/operations_spec.rb]: entered by mistake?

File: spec/vcr_cassettes/Operation/Websocket_connection/should_bail_with_no_host.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[
+
+]
```
Comment (abonas) [inline spec/vcr_cassettes/Operation/Websocket_connection/should_bail_with_no_host.json:3]: missing newline?

**Comment by abonas** [comment]
I am not a fan of "fix rubocop" commits as they are relevant inside the PR and not vs. master.
also, comments in source regarding quieting rubocop are also imo not the correct approach as this is mixing source code with external code validation, which can also change later to something else, requiring touching source code for that.
Other than that and a cpl of minor comments, LGTM

**Comment by abonas** [comment]
btw you can probably close via the commit msg cpl of issues that were opened and are solved by this PR

File: spec/vcr_cassettes/Operation/Websocket_connection/should_bail_with_no_host.json (hunk 3 above)
Comment (pilhuhn) [inline spec/vcr_cassettes/Operation/Websocket_connection/should_bail_with_no_host.json:3]: Well, those cassettes were recorded via (WS)VCR, so I don't want to manually touch them.

**Comment by pilhuhn** [comment]
@abonas would you then please (squash) merge?
This PR only solves a small part of the opened issues.

**Comment by abonas** [comment]
> @abonas would you then please (squash) merge?
> This PR only solves a small part of the opened issues.

yes, I know, you can also close them manually with a link if you prefer.
the button above afaik is only merging as is. how do you squash & merge?

**Comment by pilhuhn** [comment]
Once you click the "Merge" button once, there should (iirc) be a pull down to differentiate between the two. At least that is what the settings say :)

**Comment by abonas** [comment]
> Once you click the "Merge" button once, there should (iirc) be a pull down to differentiate between the two. At least that is what the settings say :)

I literally had a revelation now - I never noticed this option!

**Pull request merged**